### DISCUSSION TITLE
Getting GET 500 internal server error for non authenticated pages in nextjs application

### DISCUSSION THREAD

**Discussion by PriyankaManickkam** [vercel/next.js#70683] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
Binaries:
  Node: 20.16.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.3
  eslint-config-next: 14.0.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.2
Next.js Config:
  output: N/A
```

### Which example does this report relate to?

IN Nextjs

### What browser are you using? (if relevant)

Chrome 128.0.6613.85

### How are you deploying your application? (if relevant)

Other Platform

### Describe the Bug

After deployment we are getting GET 500 internal server error for specific pages in console not in all the pages. The pages which was authenticated are not having that GET 500 internal server error. Not able to reproduce that in dev mode. Even tried creating empty page that was also getting 500 internal server error. Because of this error pages are getting refreshed when we try to navigate to some other page also redux are getting reseted. 

In local also getting the same issue on build mode
<img width="959" alt="image" src="https://github.com/user-attachments/assets/64b6c2fb-387f-44bd-98e7-ac92f3a71824">

<img width="953" alt="image" src="https://github.com/user-attachments/assets/e261d897-b8ca-4e90-b999-a12e92eddb6e">

### Expected Behavior

We should not get 500 internal server error  on console. Also pages should not refresh on navigation

### To Reproduce

I don't think so others will reproduce the issue because i guess it's something with our project

**Comment by samcx** [comment]
@PriyankaManickkam Moving this to discussions because it's unlikely a Next.js issue but something with your setup!
